hw/xbox/xid-sdl.c

```diff
@@ -464,6 +479,66 @@ static void usb_xid_class_initfn(ObjectClass *klass, void *data)
     uc->handle_attach  = usb_desc_attach;
 }
 
+static HapticEmulationMode select_haptic_emulation_mode(SDL_Haptic *haptic)
+{
+    unsigned int query_result = SDL_HapticQuery(haptic);
+    if (query_result & SDL_HAPTIC_LEFTRIGHT) {
+        return EMU_HAPTIC_LEFT_RIGHT;
+    }
+
+    fprintf(stderr, "No supported haptic mode found (SDL_HapticQuery result: 0x%x). Haptic disabled\n", query_result);
+
+    return EMU_NONE;
+}
+
+static int haptic_init(USBXIDState *s)
+{
+    if (SDL_HapticOpened(s->device_index)) {
+        fprintf(stderr, "Joystick haptic effect already opened, haptic effect disabled\n");
```

Comment:
nit: This says "haptic effect disabled". We have a similar case for when no mode was found [a couple lines above] that uses "haptic disabled". I think the second case without effect is right, because no effect is involved here.